ArcGISRuntimeSDKQt_CppSamples/EditData/SnapGeometryEdits/SnapGeometryEdits.cpp

```diff
@@ -0,0 +1,335 @@
+  {
+    for (Layer* layer : *m_map->operationalLayers())
+    {
+      connect (layer, &Layer::doneLoading, this, [this]()
+      {
+        // Enable snap settings after layers load
+        if (m_layersLoaded == false)
+        {
+          m_layersLoaded = true;
+          emit layersLoadedChanged();
+        }
+      });
+    }
+  });
+}
+
+SnapGeometryEdits::~SnapGeometryEdits() = default;
+
+void SnapGeometryEdits::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<SnapGeometryEdits>("Esri.Samples", 1, 0, "SnapGeometryEditsSample");
+}
+
+MapQuickView* SnapGeometryEdits::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void SnapGeometryEdits::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_map->loadSettings()->setFeatureTilingMode(FeatureTilingMode::EnabledWithFullResolutionWhenSupported);
+
+  m_mapView->graphicsOverlays()->append(m_graphicsOverlay);
+
+  // Set the geometry editor on the map view
+  m_mapView->setGeometryEditor(m_geometryEditor);
+  m_mapView->setMagnifierEnabled(true);
+
+  emit mapViewChanged();
+  createInitialSymbols();
+  createConnections();
+}
+
+// Create symbols used by all graphics
+void SnapGeometryEdits::createInitialSymbols()
+{
+  m_pointSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Square, QColor(255, 45, 0), 10, this);
+  m_multiPointSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Square, QColor(255, 45, 0), 10, this);
+  m_lineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor(255, 45, 0), 2, this);
+  m_polygonSymbol = new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor(255, 0, 0, 75),
+                      new SimpleLineSymbol(SimpleLineSymbolStyle::Dash, QColor(0, 0, 0), 1.0, this), this);
+}
+
+void SnapGeometryEdits::createConnections()
+{
+  // Allow user to edit existing graphics by clicking on them
+  connect (m_mapView, &MapQuickView::mouseClicked, this, [this](const QMouseEvent& mouseEvent)
+  {
+    if (!m_geometryEditor->isStarted())
+    {
+      m_mapView->identifyGraphicsOverlayAsync(m_graphicsOverlay, mouseEvent.position(), 10 ,false).then(this, [this](IdentifyGraphicsOverlayResult* result)
+      {
+        // Handle editing selected graphics, if any
+        auto identifyResult = std::unique_ptr<IdentifyGraphicsOverlayResult>(result);
+        // Return if no graphics were identified
+        if (identifyResult->graphics().isEmpty())
+          return;
+
+        m_editingGraphic = identifyResult->graphics().first();
+
+        // Hide the graphic currently being edited
+        m_editingGraphic->setVisible(false);
+
+        // Start the geometry editor with the graphic's geometry
+        m_geometryEditor->start(m_editingGraphic->geometry());
+
+        emit geometryEditorStartedChanged();
+
+        result->deleteLater();
```

Comment:
we use the unique pointer above, and call deleteLater here. We should pick one pattern and go with it

Reply:
removed